pkg/database/openshiftclusters.go

```diff
@@ -289,6 +290,16 @@ func (c *openShiftClusters) EndLease(ctx context.Context, key string, provisioni
 	}, nil)
 }
 
+func (c *openShiftClusters) QueryAll(ctx context.Context, partitionKey string) (*api.OpenShiftClusterDocuments, error) {
+	docs, err := c.c.QueryAll(ctx, partitionKey, &cosmosdb.Query{
+		Query: `SELECT * FROM OpenShiftClusters doc`,
```

Comment:
Yikes, let's not do this - anyone could submit a whole pile of requests to the RP, each of which would cause the RP to pull all of or a lot of the DB.  Create two specialised functions with WHERE clauses that match each key in question and use them?

Reply:
I see, when you said `2 specialized functions` you mean still doing `QueryAll` ? 

Something like `SELECT * FROM OpenShiftClusters doc WHERE ClusterResourceGroupIDKey=XXX` and  `SELECT * FROM OpenShiftClusters doc WHERE ClientIDKey =XXX` ? 
I guess i don't need to do a `SELECT *` and just do something like `SELECT id...` ?

Also why doing 2 `QueryAll` instead of 2 simple `Get` ?

Finally out of curiosity, `each of which would cause the RP to pull all of or a lot of the DB` : I guess this is because returning the whole documents, with all the fields, is more expensive compared to just 1 dedicated document ? We just need to check is this is already present after all ?

Thanks for you time :)

Reply:
Yes, still using QueryAll.  Can't use Get because that requires the record ID and we don't know the ID of the colliding record.
`SELECT *` rather than `SELECT id` for now is simpler because it maintains the general code invariant that we deal in documents and all their fields are always populated - less thought required.

Yes, returning lots of documents is more expensive - Cosmos RUs are largely based on backend IO load and if we exhaust the RU quota other requests will have to retry.  Returning lots of docs here could make it easier for an adversary to exhaust our quota if they wanted to.